File: B.Tech/SEM - 5/GT/DFS.java
import java.util.ArrayList;
import java.util.Scanner;
import java.util.Stack;

class DFS {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter Number of nodes : ");
        int n = sc.nextInt();

        int[][] graph = new int[n][];
        ArrayList<Integer> check = new ArrayList<Integer>();
        

        for (int i = 0; i < n; i++) {
            System.out.print("Enter Number of edges : ");
            int m = sc.nextInt();
            int[] node = new int[m];
            for (int j = 0; j < m; j++) {
                node[j] = sc.nextInt();
            }
            graph[i] = node;
        }

        System.out.print("Enter starting position : ");
        int start = sc.nextInt();

        Stack<Integer> stack = new Stack<Integer>();
        stack.push(start); 
        
        while (check.size() < n) {
            int lastElement = stack.pop();
            check.add(lastElement);
            for(int i=0 ; i<graph[lastElement].length ; i++){
                if (!check.contains(graph[lastElement][i])) {
                    stack.push(graph[lastElement][i]);
                }
            }
        }

        System.out.print("Check Array : ");
        for(int i : check){
            System.out.print(i+", ");
        }
        sc.close();
    }
}
//stack : 
//check : 2,1,4

File: B.Tech/SEM - 5/DAA/DFS.c
#include <Stdio.h>
#define V 5

void DFS(int graph[V][V], int visited[V], int key){
    visited[key] = 1;
    printf("%d", key+1);
    for (int i = 0; i < V; i++){
        if(visited[i] == 0 && graph[key][i] != 0){
            DFS(graph, visited, i);
        }
    }
}

void main(){
    int graph[V][V] ={
        {0, 1, 0, 0, 1},
        {1, 0, 1, 0, 0},
        {0, 1, 0, 1, 0},
        {0, 0, 1, 0, 1},
        {1, 0, 0, 1, 0}
    };
    int visited[V] = {0};
    DFS(graph, visited, 0);
}